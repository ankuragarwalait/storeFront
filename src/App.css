/* Global Element Styles ------------------------------------------------------------------------------*/
body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

.App {
  text-align: center;
}

hr {
  margin: 1rem 0;
}

img.img-responsive {
  width: 100%;
  max-width: 100%;
}

.btn {
  display: block;
  padding: 1rem 3rem;
  font-size: 0.75rem; 
  color: white;
  border-width: 0;
  cursor: pointer;
  text-decoration: none;
}

.btn-primary {
  background-color: black;
}

.btn-secondary {
  background-color: grey;
}

.btn-tertiary {
  background-color: whote;
  color: black;
  border: 2px solid black;
}

.caret-link {
  cursor: pointer;
}

.caret-link::after {
  content: "";
  display: inline-block;
  margin-left: 0.2rem;
  vertical-align: middle;
  border-left: 0.2rem solid transparent;
  border-right: 0.2rem solid transparent;
  border-top: 0.2rem solid black;
}

/* Grid Styles ------------------------------------------------------------------------------ */

.container {
  max-width: 75rem;
  margin: 0 auto;
}

.row{
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  justify-content: space-between;
}

.col-1 {
  flex: 1 0 8.33%;
}

.col-2 {
  flex: 1 0 16.66%;
}

.col-4 {
  flex: 1 0 30%;
  margin-right: 3.33%;
}

.col-6 {
  flex: 1 0 50%;
}

.col-8 {
  flex: 1 0 66.66%;
}

.col-4:nth-of-type(3n+3) {
  margin-right: 0;
}

/* Alignment Styles ------------------------------------------------------------------------------*/

.justify-center {
  justify-content: center;
}

.align-center {
  align-items: center;
}

.justify-around {
  justify-content: space-around;
}

.text-left {
  text-align: left;
}

.text-right {
  text-align: right;
}

/* Typography and Theme ------------------------------------------------------------------------------*/

.t-price, .t-brand {
  color: gray;
}

.t-brand {
  font-size: .75rem;
  font-weight: 500;
}

.t-font-xs {
  font-size: .25rem;
}

.t-font-s {
  font-size: .5rem;
}

.t-font-m {
  font-size: .75rem;
}

.t-cancel {
  float: right;
  cursor: pointer;
  color: gray;
  font-weight: bold;
}

.t-border-grey {
  border-color: grey;
}

.t-border-light-grey {
  border-color: lightgrey;
}

.t-color-black {
  color: black;
}

.t-color-white {
  color: white;
}

.t-color-grey {
  color: grey;
}

.t-color-lightgrey {
  color: lightgrey;
}
