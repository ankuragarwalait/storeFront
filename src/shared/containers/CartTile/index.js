import React, { Component } from 'react';
import { connect } from 'react-redux';
import config from '../../../config';
import QtyWidget from '../QtyWidget';
import PropTypes from 'prop-types';
import { removeFromCart, updateCartQuantity } from '../../../store/actions/cartActions';

class CartTile extends Component {
    static propTypes = {
        prodItem: PropTypes.shape({
            id: PropTypes.string,
            title: PropTypes.string,
            image: PropTypes.string,
            brand: PropTypes.string,
            qty: PropTypes.number,
            price: PropTypes.number
        }),
        removeCartProduct: PropTypes.func,
        updateQuantity: PropTypes.func
    }
    constructor(props) {
        super(props);
        this.handleRemoveClick = this.handleRemoveClick.bind(this);
    }

    handleRemoveClick() {
        this.props.removeCartProduct(this.props.prodItem.id);
    }

    render() {
        const prodItem = this.props.prodItem;
        return (
            <div className="row">
                <div className="col-6 text-left">
                    <img className="img-responsive" src={`${config.imageURL}/${prodItem.image}`} alt={prodItem.id}/>
                    <div className="cart-prod-details">
                        <p className="t-brand">{prodItem.brand}</p>
                        <p>{prodItem.title}</p>
                    </div>
                </div>
                <div className="col-2">
                    <QtyWidget prodQty={prodItem.qty} updateQuantity={this.props.updateQuantity} shouldAutoUpdate={true} prodId={prodItem.id}/>
                </div>
                <div className="col-1 t-price">
                    {`$${parseFloat(prodItem.qty*prodItem.price).toFixed(2)}`}
                </div>
                <div className="col-1 text-right"><span className="t-cancel" onClick={this.handleRemoveClick}>&#10005;</span></div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        removeCartProduct: (prodId) => dispatch(removeFromCart(prodId)),
        updateQuantity: (qty, prodId) => dispatch(updateCartQuantity(qty, prodId))
    }
}

export default connect(null, mapDispatchToProps)(CartTile)
