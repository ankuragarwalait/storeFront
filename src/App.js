import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.css';
import Category from "./category/Category";
import Cart from "./cart/Cart";
import Product from "./product/Product";
import Header from './shared/containers/Header';

class App extends Component {
  render() {
    return (
      <div className="App">    
        <Header cartCount={0} />
        <Switch>
          <Route exact path="/" component={Category} />
          <Route path="/cart" component={Cart}/>
          <Route path="/product/:id" component={Product}/>
        </Switch>
      </div>
    );
  }
}

export default App;
