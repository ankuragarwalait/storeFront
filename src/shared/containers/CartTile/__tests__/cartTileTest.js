import renderer from 'react-test-renderer';
import React from 'react';
import CartTile from '..';
import { Provider } from 'react-redux';
import { mockStore } from '../../../../App.test';

let tree = null;
let store = null;

describe('test connected cart tile component snapshot', () => {
    beforeEach(()=> {
        store = mockStore;
        const props = {
            prodItem: store.getState().cart.items["mashiko-yaki-indigo-small-plate"]
        }
        tree = renderer.create(<Provider store={store}>
                                    <CartTile {...props}/>
                                </Provider>);
    });

    it('cart tile rendered snapshot', ()=> {
        const prodJson = tree.toJSON();
        expect(prodJson).toMatchSnapshot();
    });  

    it('cart tile update quantity on decrement click', ()=> {
        tree.root.findByProps({className: 'btn-dec'}).props.onClick();
        expect(store.getState().cart.itemCount).toBe(2);
        expect(store.getState().cart.total).toBe(20);
    }); 
    
    it('cart tile remove product click', ()=> {
        tree.root.findByProps({className: 't-cancel'}).props.onClick();
        expect(Object.keys(store.getState().cart.items).length).toBe(1);
    }); 
});