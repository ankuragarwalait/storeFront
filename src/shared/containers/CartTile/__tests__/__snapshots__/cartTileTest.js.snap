// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test connected cart tile component snapshot cart tile rendered snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-6 text-left"
  >
    <img
      alt="mashiko-yaki-indigo-small-plate"
      className="img-responsive"
      src="/media/mashiko-yaki-indigo-small-plate.jpg"
    />
    <div
      className="cart-prod-details"
    >
      <p
        className="t-brand"
      >
        Kiriko
      </p>
      <p>
        Mashiko-Yaki Indigo Small Plate
      </p>
    </div>
  </div>
  <div
    className="col-2"
  >
    <div>
      <input
        className="counter-field"
        readOnly={true}
        value={2}
      />
      <div
        className="counter-control"
      >
        <button
          onClick={[Function]}
        >
          +
        </button>
        <button
          className="btn-dec"
          onClick={[Function]}
        >
          -
        </button>
      </div>
    </div>
  </div>
  <div
    className="col-1 t-price"
  >
    $20.00
  </div>
  <div
    className="col-1 text-right"
  >
    <span
      className="t-cancel"
      onClick={[Function]}
    >
      âœ•
    </span>
  </div>
</div>
`;
